modules/programs/eww.nix

```diff
@@ -23,12 +21,74 @@ in {
       '';
     };
 
-    configDir = mkOption {
-      type = types.path;
-      example = literalExpression "./eww-config-dir";
+    yuck = {
+      enable = mkEnableOption ""
+      mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = ''
+          (defwindow example
+            :monitor 0
+            :geometry (geometry :x "0%"
+                                 :y "20px"
+                                 :width "90%"
+                                 :height "30px"
+                                 :anchor "top center")
+            :stacking "fg"
+            :reserve (struts :distance "40px" :side "top")
+            :windowtype "dock"
+            :wm-ignore false
+          "example content")
+        '';
+        description = ''
+          The content that gets symlinked to
+          {file} `$XDG_CONFIG_HOME/eww/eww.yuck`.
+        '';
+      };
+    };
+
+    scss = {
+      enable = mkEnableOption "Enables string based scss config";
+      config = mkOption {
+        type = types.nullOr types.lines;
+        default = null;
+        example = ''
+          window {
+            background: pink;
+          }
+        '';
+        description = ''
+          The directory that gets symlinked to
+          {file} `$XDG_CONFIG_HOME/eww/eww.scss`.
+        '';
+      };
+    };
+
+    configDir = {
```

Comment:
I'm not fond of this option and would suggest removing it. if a user really wants to do this I think they can set `xdg.configFile.eww.source` directly…

Reply:
Hehe, just noticed this option actually being present before. In that case I guess we have to keep it 😀